Lib/test/test_zipfile.py

```diff
@@ -2403,10 +2403,19 @@ def add_dirs(zipfile):
     any directories implied by the presence of children.
     """
     names = zipfile.namelist()
+    subdirs = set([
```

Comment:
This whole function (`add_dirs`) looks suspiciously-similar to the code in `zipfile.Path._add_implied_dirs`. When these functions were simple, it maybe made sense to have some duplication across them, but if they're going to grow all of this new behavior, it probably makes sense to extract the common logic and re-use it.

Reply:
On the other hand, it might be important to keep these separate so as to avoid the tests and implementation being over-matched. Really what we need is a way to "determine implied dirs" from a set of names, have a test for that functionality, and then re-use that function.

Reply:
I feel like the function (``add_dirs``) is not needed as _add_implied_dirs does exactly the same thing. ``add_dirs`` is also only used in the tests. I think we should just remove add_dirs altogether. What do you think?

Reply:
`add_dirs` is used in the tests to create two variants of a zip file, one with implicit dirs and one with explicit dirs. `zipfile.Path` is designed to work with both, so it's important that the tests exercise both variants.

Reply:
@jaraco - based on my latest update where I added a standalone ``_parents`` method, Should I add tests for ``_add_implied_dirs``?

Reply:
I'm not sure I fully understand the question, but I don't believe the code needs tests for internal behavior as long as the user-facing behavior is exercised.